FluidNC/src/Machine/MachineConfig.cpp

```diff
@@ -35,6 +35,8 @@ namespace Machine {
         handler.item("name", _name);
         handler.item("meta", _meta);
 
+        handler.extSection("uart_", &Uart::externals);
+
```

Comment:
It would appear that "uart_" is a regex pattern that is intended to match "uart1" etc.  I don't see how that works, since _ is not a regex special character.  I would have expected the pattern to be "uart.".

Reply:
Sorry, you are right. I switched to `uart_first` instead of `uart_1`. The idea was to be able to use any name you like as long as it starts with `uart_`, e.g. `uart_xy` and `uart_z`.

Reply:
So the use of regular expressions is just a way of implementing "startsWith" - although the regex formulation will also match "foouart".

Reply:
I am trying to figure out a way to use the GenericFactory mechanism for this, like for Spindles and Motors and Kinematic systems.

Reply:
I knew that this is your longterm plan. I am just not sure how to do it nicely together with what is already there. Of course you could also define a `extSection` with `^motor\d` and storing it to `&Motor::external`.
I just do not like those static external lists in the objects. But my knowledge about c/c++ is very limited.